packages/core/admin/shared/contracts/api-token.ts

```diff
@@ -0,0 +1,97 @@
+export type ApiTokenPermission = {
+  id: number | `${number}`;
+  action: string;
+  token: ApiToken | number;
+};
+
+export type ApiToken = {
+  id?: number | `${number}`;
+  name: string;
+  description: string;
+  accessKey: string;
+  lastUsedAt: number;
+  lifespan: number | null;
+  expiresAt: number;
+  type: 'read-only' | 'full-access' | 'custom';
+  permissions: (number | ApiTokenPermission)[];
+};
+
+type ApiTokenBody = Pick<ApiToken, 'lifespan' | 'description' | 'type' | 'name' | 'permissions'>;
+type ApiTokenResponse = Omit<ApiToken, 'accessKey'>;
+
+/**
+ * POST /api-tokens - Create an api token
+ */
+export declare namespace Create {
+  export interface Request {
+    body: ApiTokenBody;
+  }
+
+  export interface Response {
+    data: ApiToken;
+  }
+}
```

Comment:
I have given a thought at the idea of not adding empty Request & Response attributes.

There are contracts that doesn't need query / body/ errors. So , with it's current contract definition that might be ok.
I do think the problem can appear when refactoring the types; adding new errors, new parameters...
If we do keep all of those attributes in place, frontend can still use them. And in the eventuality that one of those change, typescript would exactly say us which endpoints requests need to be refactored in the ui side.

For example here, if we were to add a new error:
```
  export interface Response {
    data: ApiToken
	errors?: errors.ApplicationError
  }
```

And the ui side has a type guard system to handle all errors, we wouldn't know.

So now I am really inclined to always use body/query , data/errors for the contract types. wdyt?